metricsbp/statsd.go

```diff
@@ -124,14 +98,17 @@ func NewStatsd(ctx context.Context, cfg Config) *Statsd {
 	}
 	st.ctx, st.cancel = context.WithCancel(ctx)
 
+	var sink io.Writer
```

Comment:
you can do

```
sink := io.Discard
if cfg.Endpoint != "" {
  sink = ...
}
```

Reply:
this is 3-state, not 2-state. I can do that but I would still need:

```go
sink := io.Discard
if cfg.Endpoint != "" {
  sink = ...
} else if cfg.NoBlackhole {
  sink = nil
}
```
otherwise the `if` on line 107 will stop working.